### DISCUSSION TITLE
Stratification of multilabel data

### DISCUSSION THREAD

**Discussion by dokato** [embeddings-benchmark/mteb#698] [General]
Recently, new Multilabel classification task has been added: https://github.com/embeddings-benchmark/mteb/pull/440 

Looking at datasets available on HF, they're typically quite large, e.g. >50k examples. What's the best way to train/test split it, or just sample it?

Maybe sth from: http://scikit.ml/stratification.html ?

cc @x-tabdeveloping as you added that in

**Comment by x-tabdeveloping** [comment]
I have added some stratified subsampling for multilabel data in this PR: #694 You can have a look at the code there.
But in essence I think you can just use scikit-learn's `train_test_split` with an encoded labels array as the `stratify` argument.

Something like this:
```python
from sklearn.model_selection import train_test_split
from sklearn.preprocessing import MultiLabelBinarizer

X: ArrayLike = [...]
labels: list[list[str]] = [...]
encoded_labels = MultiLabelBinarizer.fit_transform(labels)

X_train, y_train, X_test, y_test = train_test_split(X, labels, stratifiy=encoded_labels)
```

**Reply by dokato** [reply]
Great, thanks! I must have overlooked that.

**Reply by x-tabdeveloping** [reply]
It's not merged yet, so you haven't overlooked anything :D
I'm not sure how well it works though, a lot of times it throws a warning and doesn't reduce the dataset size, so we might have to look into it.

**Reply by dokato** [reply]
Yea, because on surface it doesn't seem like such a trivial problem, especially when dealing with narrowly represented classes.

**Reply by dokato** [reply]
@x-tabdeveloping have a look at quick and dirty attempt of using `iterative_train_test_split` from this `scikit-multilearn` library: 
https://github.com/embeddings-benchmark/mteb/pull/546
It works quite neat and fast, though requires a few hack arounds to make it running with our flow, plus we need to add another dependency for just this one function, LMK what you think...

**Reply by x-tabdeveloping** [reply]
@dokato Can you make this in a separate PR and then we can discuss it there? I have some thoughts about it

**Reply by dokato** [reply]
DIscussion moved to: https://github.com/embeddings-benchmark/mteb/pull/760
